geomstats/backend/tensorflow.py

```diff
@@ -0,0 +1,136 @@
+"""Tensorflow based computation backend."""
+
+# TODO(johmathe): Reproduce all unit tests with tensorflow backend.
+
+import tensorflow as tf
+
+pi = tf.pi
+
+
+def abs(x):
+    return tf.abs(x)
+
+
+def zeros(x):
+    return tf.zeros(x)
+
+
+def ones(x):
+    return tf.ones(x)
+
+
+def sin(x):
+    return tf.sin(x)
+
+
+def cos(x):
+    return tf.cos(x)
+
+
+def cosh(x):
+    return tf.cosh(x)
+
+
+def sinh(x):
+    return tf.sinh(x)
+
+
+def tanh(x):
+    return tf.tanh(x)
+
+
+def arccosh(x):
+    return tf.arccosh(x)
+
+
+def tan(x):
+    return tf.tan(x)
+
+
+def arcsin(x):
+    return tf.asin(x)
+
+
+def arccos(x):
+    return tf.acos(x)
+
+
+def shape(x):
+    return tf.shape(x)
+
+
+def dot(x, y):
+    return tf.reduce_sum(tf.multiply(x, y))
+
+
+def maximum(x, y):
+    return tf.maximum(x, y)
+
+
+def greater_equal(x, y):
+    return tf.greater_equal(x, y)
+
+
+def to_ndarray(element, to_ndim, axis=0):
+
+    if element.ndim == to_ndim - 1:
+        element = tf.expand_dims(element, axis=axis)
+    assert element.ndim == to_ndim
+    return element
+
+
+def sqrt(x):
+    return tf.sqrt(x)
+
+
+def norm(x, axis=None, keepdims=None):
+    return tf.linalg.norm(x, axis=axis, keepdims=keepdims)
+
+
+def isclose(x, y, rtol=1e-05, atol=1e-08):
+    rhs = tf.constant(atol) + tf.constant(rtol) * tf.abs(y)
+    return tf.less_equal(tf.abs(tf.sub(x, y)), rhs)
+
+
+def allclose(x, y, rtol=1e-05, atol=1e-08):
+    return tf.reduce_all(isclose(x, y, rtol=rtol, atol=atol))
+
+
+def less_equal(x, y):
+    return tf.less_equal(x, y)
+
+
+def eye(N, M=None):
```

Comment:
Why uppercases here?

Reply:
taking the same parameters naming as numpy.